### ISSUE TITLE
Allow newlines after declaration colon

### ISSUE THREAD

**Issue by ntwb** [stylelint/stylelint#386]
I've got quite a bit off CSS in the following format that I'd like to able to parse as valid

```css
.wp-core-ui .button.focus,
.wp-core-ui .button:focus,
.wp-core-ui .button-secondary:focus,
.wp-core-ui .button-link:focus {
	-webkit-box-shadow:
		0 0 0 1px #5b9dd9,
		0 0 2px 1px rgba(30, 140, 190, 0.8);
	box-shadow:
		0 0 0 1px #5b9dd9,
		0 0 2px 1px rgba(30, 140, 190, 0.8);
}
```

Currently the following warning is generated from CSS such as this:
```
165:21	⚠  Expected single space after ":" (declaration-colon-space-after) [stylelint]
168:13	⚠  Expected single space after ":" (declaration-colon-space-after) [stylelint]
```
Any suggestions on the best way to allow this?

**Comment by ntwb** [comment]
via https://make.wordpress.org/core/handbook/best-practices/coding-standards/css/#values

> _"Newlines should be used for lengthier multi-part values such as those for shorthand properties like box-shadow and text-shadow. Each subsequent value after the first should then be on a new line, indented to the same level as the selector and then spaced over to left-align with the previous value."_

**Comment by davidtheclark** [comment]
Added `declaration-colon-newline-after` rule. You'll probably want to use the `"always-multi-line"` option, right?

**Comment by ntwb** [comment]
I've added this new rule `"declaration-colon-newline-after": [ 2, "always-multi-line" ],` to our ruleset in [stylelint/stylelint-config-wordpress/blob/master/index.js](https://github.com/stylelint/stylelint-config-wordpress/blob/master/index.js)

Using the same CSS per above with a tweak to force the new rule to display a warning:
```css
.wp-core-ui .button.focus,
.wp-core-ui .button:focus,
.wp-core-ui .button-secondary:focus,
.wp-core-ui .button-link:focus {
	-webkit-box-shadow:
		0 0 0 1px #5b9dd9,
		0 0 2px 1px rgba(30, 140, 190, 0.8);
	box-shadow:0 0 0 1px #5b9dd9,
		0 0 2px 1px rgba(30, 140, 190, 0.8);
}
```
Displays the following warnings:
```
165:21	⚠  Expected single space after ":" (declaration-colon-space-after) [stylelint]
168:12	⚠  Expected newline after ":" (declaration-colon-newline-after) [stylelint]
168:13	⚠  Expected single space after ":" (declaration-colon-space-after) [stylelint]
```

A slight tweak again to add the space after the colon on line 168 but not a new line:
```css
.wp-core-ui .button.focus,
.wp-core-ui .button:focus,
.wp-core-ui .button-secondary:focus,
.wp-core-ui .button-link:focus {
	-webkit-box-shadow:
		0 0 0 1px #5b9dd9,
		0 0 2px 1px rgba(30, 140, 190, 0.8);
	box-shadow: 0 0 0 1px #5b9dd9,
		0 0 2px 1px rgba(30, 140, 190, 0.8);
}
```
Results in
```
165:21	⚠  Expected single space after ":" (declaration-colon-space-after) [stylelint]
168:12	⚠  Expected newline after ":" (declaration-colon-newline-after) [stylelint]
```

So regardless of the new rule definition `declaration-colon-newline-after` a warning is still being displayed for `declaration-colon-space-after`.

So an exception or ignore condition for the rule `declaration-colon-space-after` is needed to accommodate when `"declaration-colon-newline-after": [ 2, "always-multi-line" ],` is defined.

For an exception:
```js
    "declaration-colon-space-after": [ 2, "always", {
      except: ["multi-line"],
    } ],
```
Or for an ignore:
```js
    "declaration-colon-space-after": [ 2, "always", {
      ignore: ["multi-line"],
    } ],
```

**Comment by davidtheclark** [comment]
Ah, right. I believe the way that we've been handling this so far would be to add to `declaration-colon-space-after` a primary option for `"always-single-line"`. So you would have configuration like this

**Comment by ntwb** [comment]
> Does that sound like it would work?
Yes :smile:

**Comment by ntwb** [comment]
:+1:
